napari/__main__.py

```diff
@@ -187,5 +192,65 @@ def main():
             )
 
 
+def _run_pythonw(python_path):
+    """Execute this script again through pythonw.
+
+    This can be used to ensures we're using a framework
+    build of Python on macOS which fixes menubar issues.
+
+    Parameters
+    ----------
+    python_path : pathlib.Path
+        Path to python framework build.
+    """
+    import subprocess
+
+    cwd = Path.cwd()
+    cmd = [python_path, '-m', 'napari']
+    env = os.environ.copy()
+
+    # Append command line arguments.
+    if len(sys.argv) > 1:
+        cmd.append(*sys.argv[1:])
+
+    result = subprocess.run(cmd, env=env, cwd=cwd)
+    sys.exit(result.returncode)
+
+
+def main():
+    # Ensure we're always using a "framework build" on the latest
+    # macOS to ensure menubar works without needing to refocus napari.
+    # We try this for macOS later than the Catelina release
+    # See https://github.com/napari/napari/pull/1554 and
+    # https://github.com/napari/napari/issues/380#issuecomment-659656775
+    # and https://github.com/ContinuumIO/anaconda-issues/issues/199
+    _MACOS_LATEST = sys.platform == "darwin" and StrictVersion(
+        platform.release()
+    ) > StrictVersion('19.0.0')
+    _RUNNING_CONDA = "CONDA_PREFIX" in os.environ
+    _RUNNING_PYTHONW = "PYTHONEXECUTABLE" in os.environ
+
+    if _MACOS_LATEST and _RUNNING_CONDA and not _RUNNING_PYTHONW:
+        python_path = Path(sys.exec_prefix) / 'bin' / 'pythonw'
+
+        if python_path.exists():
+            # Running again with pythonw will exit this script
+            # and use the framework build of python.
+            _run_pythonw(python_path)
+        else:
+            msg = (
+                'pythonw executable not found. '
+                'The menubar might only become '
+                'functional on the macOS after '
+                'focus is toggled in and out of '
+                'napari. To fix this problem, '
+                'please install python.app via conda, '
+                'which can be done with '
+                '`conda install -c conda-forge python.app`.'
+            )
+            raise warnings.warn(msg)
```

Comment:
```suggestion
           warnings.warn(msg)
```

Oops! This should not be a raise! Here's what happens when I run `python -m napari` without python.app installed:

```
 ~/projects/napari (fix_mac_menubar)
 $ python -m napari
/Users/jni/projects/napari/napari/__main__.py:251: UserWarning: pythonw executable not found. The menubar might only become functional on the macOS after focus is toggled in and out of napari. To fix this problem, please install python.app via conda, which can be done with `conda install -c conda-forge python.app`.
  raise warnings.warn(msg)
Traceback (most recent call last):
  File "/Users/jni/conda/envs/all/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/jni/conda/envs/all/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/jni/projects/napari/napari/__main__.py", line 256, in <module>
    sys.exit(main())
  File "/Users/jni/projects/napari/napari/__main__.py", line 251, in main
    raise warnings.warn(msg)
TypeError: exceptions must derive from BaseException
```